File: JavaClient/src/main/java/com/security/crypto/Configuration/CiphersForUse.java
package com.security.crypto.Configuration;


public class CiphersForUse {
    private String CipherAlgorithm;
    private String HashAlgorithm;


    public CiphersForUse() {
        this.CipherAlgorithm = null;
        this.HashAlgorithm = null;
    }

    public CiphersForUse(String CipherAlgorithm, String HashAlgorithm) {
        this.CipherAlgorithm = CipherAlgorithm;
        this.HashAlgorithm = HashAlgorithm;
    }

    public void setHashAlgorithm(String hashAlgorithm) {
        HashAlgorithm = hashAlgorithm;
    }

    public void setCipherAlgorithm(String cipherAlgorithm) {
        CipherAlgorithm = cipherAlgorithm;
    }

    public String getHashAlgorithm() {
        return HashAlgorithm;
    }

    public String getCipherAlgorithm() {
        return CipherAlgorithm;
    }

    @Override
    public String toString() {
        return super.toString();
    }
}

File: SecureUWPClient/SecureUWPClient/Configuration/CiphersForUse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Security.Cryptography.Core;

namespace SecureUWPClient.Configuration
{
   public class CiphersForUse
    {
        private String cipherAlgorithm;
        private String hashAlgorithm;


        public CiphersForUse()
        {
            this.cipherAlgorithm = null;
            this.hashAlgorithm = null;
        }

        public CiphersForUse(String cipherAlgorithm, String hashAlgorithm)
        {
            this.cipherAlgorithm = cipherAlgorithm;
            this.hashAlgorithm = hashAlgorithm;
        }


        public string CipherAlgorithm
        {
            get
            {
                return cipherAlgorithm;
            }
            set
            {
                cipherAlgorithm = value;
            }
        }


        public string HashAlgorithm
        {
            get
            {
                if (hashAlgorithm.Contains("256"))
                    return MacAlgorithmNames.HmacSha256;
                else if(hashAlgorithm.Contains("SHA1"))
                    return MacAlgorithmNames.HmacSha1;
                else
                    return MacAlgorithmNames.HmacMd5;
            }
            set
            {
                hashAlgorithm = value;
            }
        }

        public override string ToString()
        {
            return base.ToString() + ": ";
        }
    }
}
